14 feature/select2 integration

rajanvijayan opened the pull request.

rajanvijayan committed:
Add select2 for move post category section

rajanvijayan committed:
Remove js for gitignore list and add select2 js and select 2 css

rajanvijayan committed:
Revert gitignore list

rajanvijayan committed:
Add select2 in Move post by tags section

rajanvijayan committed:
Add select2 in Move posts by custom taxonomies section

rajanvijayan committed:
Add select 2 for Tags to category and Category to Tags sections

rajanvijayan committed:
Update typo issue for text domain names

rajanvijayan committed:
Fix typo issue

Fix #14

sudar commented on assets/js/bulk-move.js:
```diff
@@ -140,49 +176,40 @@ jQuery(document).ready(function () {
-			});
+		jQuery( 'tr.term-select-row' ).show();
+
+		jQuery( '#smbm_mbct_selected_term, #smbm_mbct_mapped_term' ).select2({
```
Looks like some whitespace changes have gotten into. Revert those changes since they are adding noise to the diff.

sudar commented on bulk-move.php:
```diff
@@ -69,10 +69,39 @@ function load_bulk_move() {
 	$bulk_move->set_plugin_file( __FILE__ );
 
 	add_action( 'plugins_loaded', array( $bulk_move, 'load' ), 101 );
+
```
This should be moved inside the `get_ajax_handlers` function. Also the entire AJAX handler should be made into a class. Check `BM_Request_CustomTaxonomyAction` class

sudar commented on bulk-move.php:
```diff
@@ -69,10 +69,39 @@ function load_bulk_move() {
+	/**
+	 * This is Ajax hook, It's runs when user search categories or tags on bulk-delete-posts page
+	 *
+	 * @since 5.7.0
```
This is wrong. The version of Bulk Move will be v2.0.0

sudar commented on include/Metabox/Posts/Category.php:
```diff
@@ -31,26 +31,38 @@ public function render() {
-							'orderby'      => 'NAME',
-							'hide_empty'   => false,
-						) );
+						$categories = get_categories( array(
```
Don't duplicate code. Create this as a helper function in the `BM_Metabox_PostBase` class

sudar commented on include/Metabox/Posts/CategoryToTag.php:
```diff
@@ -30,19 +30,40 @@ public function render() {
-							'hide_empty'   => false,
-						) );
-						?>
+                    	<?php
```
Move this to the helper function.

sudar replied:
You are now missing some of the parameters that we had earlier to the `wp_dropdown_categories` function.

sudar replied:
@rajanit2000 This comment is not addressed.

sudar commented on include/Metabox/Posts/CustomTaxonomy.php:
```diff
@@ -26,7 +26,6 @@ public function render() {
 					<td scope="row" colspan="2">
 						<?php _e( 'Select the post type to show its custom taxonomy.', 'bulk-move' ); ?>
 					</td>
-					<td scope="row">
```
Any specific reason for removing these `td`s?

rajanvijayan replied:
Its unwanted td and its not properly closed

sudar replied:
👍

sudar commented on include/Metabox/Posts/Tag.php:
```diff
@@ -30,11 +30,37 @@ public function render() {
+							<?php } ?>
+							</select>
+						<?php }
+						// $this->render_tags_dropdown( 'smbm_mt_new_tag', $tags, true ); ?>
```
Don't leave commented code.

sudar commented on package.json:
```diff
@@ -18,7 +18,9 @@
     "url": "https://github.com/sudar/bulk-move.git"
   },
-  "devDependencies": {},
+  "devDependencies": {
```
Any reason for adding grunt as a dependency?

sudar commented:
@rajanit2000 

Bulk Move is more object oriented than Bulk Delete. You need to make your changes to fit the existing class patterns.

The `select2` library should not be checked-in directly. It should be included using bower and then included in the build process. See how it is done in Bulk Delete and then replicate it here.

There are also some code style issues. Check the result from styleci to fix those issues.

Also implement all the applicable review comments from Bulk Delete.

rajanvijayan committed:
Remove white spaces

rajanvijayan committed:
Add LoadTaxonomyTerm class

rajanvijayan committed:
Update ajax action prefix

Fix #14

rajanvijayan committed:
Update StyleCI issue

Fix #14

rajanvijayan committed:
Update StyleCI issue

rajanvijayan committed:
Remove unwanted empty line

rajanvijayan committed:
Fix StyleCI

rajanvijayan commented:
@sudar i have done this changes

sudar commented on assets/css/select2.min.css:
```diff
@@ -0,0 +1 @@
+.select2-container{box-sizing:border-box;display:inline-block;margin:0;position:relative;vertical-align:middle}.select2-container .select2-selection--single{box-sizing:border-box;cursor:pointer;display:block;height:28px;user-select:none;-webkit-user-select:none}.select2-container .select2-selection--single .select2-selection__rendered{display:block;padding-left:8px;padding-right:20px;overflow:hid ...
```
You have to remove this file from git since we are pull it through npm.

sudar commented on assets/js/bulk-move.js:
```diff
@@ -9,6 +9,43 @@
 /*jslint browser: true, devel: true*/
 /*global BULK_MOVE, jQuery, document, postboxes, pagenow, ajaxurl*/
 jQuery(document).ready(function () {
+	/**
```
The changes should be done in the `assets/js/src/bulk-move.js` file.

sudar commented on assets/js/select2.min.js:
```diff
@@ -0,0 +1 @@
+/*! Select2 4.0.5 | https://github.com/select2/select2/blob/master/LICENSE.md */!function(a){"function"==typeof define&&define.amd?define(["jquery"],a):"object"==typeof module&&module.exports?module.exports=function(b,c){return void 0===c&&(c="undefined"!=typeof window?require("jquery"):require("jquery")(b)),a(c),c}:a(jQuery)}(function(a){var b=function(){if(a&&a.fn&&a.fn.select2&&a.fn.select2.am ...
```
This file should be removed from .git

sudar commented on include/BulkMove.php:
```diff
@@ -171,8 +171,8 @@ private function load_ajax_handlers() {
 			$this->ajax_handlers[] = new BM_Request_CustomTaxonomyAction();
+			$this->ajax_handlers[] = new BM_Request_LoadTaxonomyTerm();
 		}
-
```
Why this blank line was removed?

sudar replied:
@rajanit2000 This comment is not addressed.

rajanvijayan replied:
@sudar it's by mistake i have revert it

sudar replied:
@rajanit2000 

Looks like you added blank line at the wrong place.

rajanvijayan replied:
@sudar If I add blank space its comes StyleCI error. How can I sort out this

sudar replied:
@rajanit2000 Okay. Let's leave it for now.

sudar commented on include/Metabox/Posts/Category.php:
```diff
@@ -31,26 +31,38 @@ public function render() {
+							)
+						);
+
+						if( count($categories) > 50 ){?>
```
We are getting all the categories. This will fail when there are huge number of categories.

Handle it based on how you did for Bulk Delete.

sudar commented on include/Request/LoadTaxonomyTerm.php:
```diff
@@ -0,0 +1,63 @@
+		$return = array();
+
+		$terms = get_terms( array(
+			'taxonomy'   => sanitize_text_field($_GET['term']),
```
Run `phpcs` and fix all issues in this file

sudar replied:
@rajanit2000 This comment is not addressed.

rajanvijayan replied:
@sudar I have to install phpcs package on my sublime text 3, But it does not show any warnings

sudar replied:
You can quickly test it by running the command `vendor/bin/phpcs` from root directory.

Some example of code formatting issues are

`sanitize_text_field($_GET['term']),` -> Doesn't have proper space.
`bm_load_taxonomy_term(){` -> Space needed before `{`

sudar replied:
@rajanit2000 

This is not fixed yet.

rajanvijayan replied:
@sudar you only say revert the `phpcs` changes. We can update it once we have done all things.

sudar commented:
@rajanit2000 

I have left a couple of review comments.

Also this branch now has merge conflict. Merge the latest code from `dev/2.0.0` branch into this branch to fix those merge conflicts.

rajanvijayan committed:
Remove select2.min.css and select2.min.js
Add bu;k-move.js src

rajanvijayan committed:
remove echo statement

rajanvijayan committed:
Merge branch 'dev/2.0.0' into 14-feature/select2-integration

# Conflicts:
#	assets/js/src/bulk-move.js
#	include/Page/Base.php
#	package.json

rajanvijayan committed:
Update Ajax Handlers
Update get categories and get tags arguments
Add filter for select2 max limit
Remove unwanted whitespace

Fix #14

rajanvijayan committed:
Fix style ci issues

sudar commented on include/Metabox/Posts/Category.php:
```diff
@@ -31,26 +31,40 @@ public function render() {
-							'orderby'      => 'NAME',
-							'hide_empty'   => false,
-						) );
+						$bm_select2_ajax_limit_categories = apply_filters( 'bm_select2_ajax_limit_categories', BM_Request_LoadTaxonomyTerm::BM_MAX_SELECT2_LIMIT );
```
Move the code that retrieves the categories and renders it into a seperate helper function to the `BM_Metabox_PostBase` class.

sudar commented on include/Metabox/Posts/CategoryToTag.php:
```diff
@@ -30,19 +30,40 @@ public function render() {
                     <td scope="row">
-						<?php $this->render_tags_dropdown( 'smbm_mct_tag', $tags ); ?>
+                    	<?php
+                        $tags = get_tags();
```
Move the code that retrieves the tags and renders it into a seperate helper function to the `BM_Metabox_PostBase` class.

sudar commented on include/Page/Base.php:
```diff
@@ -230,6 +230,7 @@ public function modify_admin_footer_text( $footer_text ) {
 	 */
 	public function enqueue_script() {
 		wp_enqueue_script( 'bulk-move', $this->get_plugin_dir_url() . 'assets/js/src/bulk-move.js', array( 'jquery', 'postbox' ), BM_BulkMove::VERSION, true );
+		wp_enqueue_script( 'select2', $this->get_plugin_dir_url() . 'assets/js/select2.min.js', array( 'jquery' ), '4.0.0', true );
```
Add this enqueue before `bulk-move` and then add `select2` as dependencies to `bulk-move`.

sudar commented on include/Page/Base.php:
```diff
@@ -259,6 +260,7 @@ public function enqueue_script() {
 	 */
 	public function enqueue_styles() {
 		wp_enqueue_style( 'bulk-move', $this->get_plugin_dir_url() . 'assets/css/bulk-move.css', array(), BM_BulkMove::VERSION );
+		wp_enqueue_style( 'select2', $this->get_plugin_dir_url() . 'assets/css/select2.min.css', array(), '4.0.0' );
```
Add this enqueue before `bulk-move` and then add select2 as dependencies to bulk-move.

sudar replied:
@rajanit2000 

You have not added `select` as the dependency on both `bulk-move.js` and `bulk-move.css`

rajanvijayan committed:
Update minimumInputLength on js file
Update white space issue
Update get categories and get tags
Update enque script position

Fix #14

sudar committed:
Remove sourcemap declaration from js file

rajanvijayan committed:
Add functions for get categories and tags select in BM_Metabox_PostBase class

Fix #14

rajanvijayan committed:
remove `[-//# sourceMappingURL=bulk-move.js.map-]` line on js file

rajanvijayan committed:
Merge branch '14-feature/select2-integration' of https://github.com/sudar/bulk-move into 14-feature/select2-integration

# Conflicts:
#	assets/js/src/bulk-move.js

rajanvijayan committed:
fix style ci issue

sudar commented on include/Metabox/PostBase.php:
```diff
@@ -10,5 +10,62 @@
+	 *
+	 * @param string $name Select tag's name.
+	 */
+	protected function render_tags_select( $name = '' ) {
```
Name should not be an optional parameter.  Also let's call this function as `render_tags_dropdown`.

sudar commented on include/Metabox/PostBase.php:
```diff
@@ -10,5 +10,62 @@
+	 *
+	 * @param string $name Select tag's name.
+	 */
+	protected function render_categories_select( $name = '' ) {
```
Name should not be an optional parameter. Also let's call this function as `render_categories_dropdown`.

sudar commented on include/Metabox/Posts/Tag.php:
```diff
@@ -30,11 +30,11 @@ public function render() {
 				<tr>
 					<td scope="row" >
-						<?php $this->render_tags_dropdown( 'smbm_mt_old_tag', $tags ); ?>
+						<?php $this->render_tags_select( 'smbm_mt_old_tag' );?>
```
If you look a little above this function, we have already retrieved tags.

So let's pass tags to this function and treat it as an optional parameter. If it is present then we will use it. Otherwise we can retrieve it.

Refer to the existing `reder_tags_select` function to see how to do it.

Also you may have to change the existing `render_tags_select` function instead of creating a new one.

rajanvijayan replied:
@sudar i am not sure is it correct. 

I have to remove existing `render_tags_select` function on Base.php file. Because its not used anywhere.

rajanvijayan committed:
Fix phpcs warnings 
Fix #14

rajanvijayan committed:
Add select2 js and css as dependencies
Fix #14

rajanvijayan committed:
Update rendering tags and categories functions names

rajanvijayan commented:
@sudar I have done this review changes

rajanvijayan committed:
Revert "remove existing render_tags_select function"

This reverts commit e0e0602d178319bc8910d05c16d7d1a6cdf40375.

rajanvijayan committed:
remove existing render_tags_select function
Fix #14

rajanvijayan committed:
Revert "Fix phpcs warnings Fix #14"

This reverts commit ec0f971d993074c153d426d7f316fca3de6dd72b.

rajanvijayan committed:
Update exist render_tags_dropdown function and add tag as optional parameter

sudar commented on include/Metabox/Base.php:
```diff
@@ -242,20 +242,27 @@ protected function get_tags_or_fail() {
-		</select>
-		<?php
+	protected function render_tags_dropdown( $name, $tags = array(), $show_option_none = false ) {
+		$bm_select2_ajax_limit_tags = apply_filters( 'bm_select2_ajax_limit_tags', BM_Request_LoadTaxonomyTerm::BM_MAX_SELECT2_LIMIT );
```
There is already a helper function to retrieve the tags. Use that function.

sudar commented on include/Metabox/Base.php:
```diff
@@ -242,20 +242,27 @@ protected function get_tags_or_fail() {
-			<?php endforeach; ?>
-		</select>
-		<?php
+	protected function render_tags_dropdown( $name, $tags = array(), $show_option_none = false ) {
```
Let's move this function to `BM_Metabox_PostBase` class.

sudar commented on include/Page/Base.php:
```diff
@@ -229,7 +229,8 @@ public function modify_admin_footer_text( $footer_text ) {
 	 */
 	public function enqueue_script() {
-		wp_enqueue_script( 'bulk-move', $this->get_plugin_dir_url() . 'assets/js/src/bulk-move.js', array( 'jquery', 'postbox' ), BM_BulkMove::VERSION, true );
+		wp_register_script( 'select2', $this->get_plugin_dir_url() . 'assets/js/select2.min.js' );
```
Add jquery as dependency to `select2` library and also include its version. You can find the version that we are using from the `packages.json` file.

sudar commented on include/Page/Base.php:
```diff
@@ -258,7 +259,8 @@ public function enqueue_script() {
 	 */
 	public function enqueue_styles() {
-		wp_enqueue_style( 'bulk-move', $this->get_plugin_dir_url() . 'assets/css/bulk-move.css', array(), BM_BulkMove::VERSION );
+		wp_register_style( 'select2', $this->get_plugin_dir_url() . 'assets/css/select2.min.css' );
```
Add jquery as dependency to `select2` library and also include its version. You can find the version that we are using from the `packages.json` file.

rajanvijayan committed:
Add get_tags_or_fail() function in render tags

rajanvijayan committed:
Fix styleCI issue 

Fix #14

sudar commented on include/Metabox/Base.php:
```diff
@@ -245,7 +245,7 @@ protected function get_tags_or_fail() {
-		
+
 		if ( empty( $tags ) ){
 			$tags = get_tags_or_fail();
```
The function should be called using `$this->get_tags_or_fail`

sudar reviewed (changes requested):
One more small change is required before we can merge this.

rajanvijayan committed:
update select2 enque script

Fix #14

rajanvijayan committed:
Fix get_tags_or_fail function calling 

Fix #14
